src/cudamatrix/cu-device.h

```diff
@@ -184,6 +191,7 @@ class CuDevice {
   /// (i.e. from outside the class), call this only if Enabled() returns true.
   bool IsComputeExclusive();
 
+  // Shouldn't this be a private constructor?
```

Comment:
@danpovey I think we may be breaking the Singleton nature of the CuDevice class by exposing this constructor. I'm not completely sure about the implications of this yet, but it seems like we should remove this from the public interface if possible.